From 7bcacca214b1de0d9f300f0f1bb3a1419d4c70a3 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Sat, 13 Sep 2014 16:55:12 +0200
Subject: [PATCH 17/60] cloud-print: disable in prefs and component loader

Cloud Print should normally be autodisabled if no printers are shared.
But maybe that is not enough(?) Probably also needs to gain a checkbox
in the settings. Revisit..]

Also changes the setting for notifications about new cloud printers to
be off by default.
---
 chrome/browser/extensions/component_loader.cc                 | 2 ++
 chrome/browser/ui/browser_ui_prefs.cc                         | 2 +-
 chrome/browser/ui/webui/local_discovery/local_discovery_ui.cc | 4 ----
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/chrome/browser/extensions/component_loader.cc b/chrome/browser/extensions/component_loader.cc
index c5bcc93a9be0..df1dd97648b8 100644
--- a/chrome/browser/extensions/component_loader.cc
+++ b/chrome/browser/extensions/component_loader.cc
@@ -464,9 +464,11 @@ void ComponentLoader::AddDefaultComponentExtensions(
 #else  // defined(OS_CHROMEOS)
   DCHECK(!skip_session_components);
 #if BUILDFLAG(ENABLE_PRINTING)
+#if 0 /* IRIDIUM */
   // Cloud Print component app. Not required on Chrome OS.
   Add(IDR_CLOUDPRINT_MANIFEST,
       base::FilePath(FILE_PATH_LITERAL("cloud_print")));
+#endif
 #endif  // BUILDFLAG(ENABLE_PRINTING)
 #endif  // defined(OS_CHROMEOS)
 
diff --git a/chrome/browser/ui/browser_ui_prefs.cc b/chrome/browser/ui/browser_ui_prefs.cc
index eab631779f7d..691af893f164 100644
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -72,7 +72,7 @@ void RegisterBrowserUserPrefs(user_prefs::PrefRegistrySyncable* registry) {
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterStringPref(prefs::kCloudPrintEmail, std::string());
   registry->RegisterBooleanPref(prefs::kCloudPrintProxyEnabled, true);
-  registry->RegisterBooleanPref(prefs::kCloudPrintSubmitEnabled, true);
+  registry->RegisterBooleanPref(prefs::kCloudPrintSubmitEnabled, false);
   registry->RegisterDictionaryPref(prefs::kBrowserWindowPlacement);
   registry->RegisterDictionaryPref(prefs::kBrowserWindowPlacementPopup);
   registry->RegisterDictionaryPref(prefs::kAppWindowPlacement);
diff --git a/chrome/browser/ui/webui/local_discovery/local_discovery_ui.cc b/chrome/browser/ui/webui/local_discovery/local_discovery_ui.cc
index 187bee04cb49..d3d0eeb58d57 100644
--- a/chrome/browser/ui/webui/local_discovery/local_discovery_ui.cc
+++ b/chrome/browser/ui/webui/local_discovery/local_discovery_ui.cc
@@ -127,10 +127,6 @@ void LocalDiscoveryUI::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
       prefs::kLocalDiscoveryNotificationsEnabled,
-#if defined(OS_WIN)
       false,
-#else
-      true,
-#endif
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
 }
-- 
2.19.1

